### DISCUSSION TITLE
Edgedb CLI functionality from Python

### DISCUSSION THREAD

**Discussion by theuken** [geldata/gel#4580] [Q&A] [answered]
We would like to automate the creation and deletion of an EdgeDB database using Python.
While the Python library supports adding & manipulating data, there are several tasks which seem to be possible only in the CLI - like creating or destroying an instance. It doesn't seem like really a good solution to let Python talk to the CLI.

Our system is supposed to "rebuild itself" if e.g. the database doesn't exist.
So for testing purposes we would delete the database and/or the instance.
Our Python code is supposed to check if the instance/database xyz exists.
If it does exist, we connect to it and work with the existing data.
If it does not exist we create it AND start a specific task to rebuild the content.
The key point is that we need to check if a database with the name "xyz" exists and not simply have an empty database created.. 

How can we accomplish this ?
Is there a Python library covering the functionality of EdgeDB CLI ?

**Comment by IgnisDa** [comment]
You can just call the CLI using `subprocess.run` calls. Does that not work for you?

**Comment by elprans** [comment, marked as answer]
The CLI is just another EdgeDB client, it doesn't use any internal APIs, so you can use the bindings to do everything the CLI does.

To check if the database exists, query the `sys::Database` type:

```select sys::Database filter .name = 'your_db'```.

To drop a database, run `DROP DATABASE your_db;` and to create one run `CREATE DATABASE your_db`.

**Reply by theuken** [reply]
Many thanks - so simple ...
This was too simply to think of.
